app/components/TransactionModal/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { FC, ReactNode, useState, useEffect } from "react";
+import { Trans } from "@lingui/macro";
+import { AppNotificationStatus } from "state/app";
+import {
+  AllowancesToken,
+  AllowancesSpender,
+} from "@klimadao/lib/types/allowances";
+
+import { Modal } from "components/Modal";
+import { Approve } from "./Approve";
+import { Submit } from "./Submit";
+import { Spinner } from "@klimadao/lib/components";
+
+import * as styles from "./styles";
+
+interface Props {
+  title: ReactNode;
+  value: string;
+  token: AllowancesToken;
+  spender: AllowancesSpender;
+  onCloseModal: () => void;
+  onApproval: () => void;
+  onSubmit: () => void;
+  status?: AppNotificationStatus | null;
+  onResetStatus: () => void;
+  hasApproval: boolean;
+}
+
+export const TransactionModal: FC<Props> = (props) => {
+  const [view, setView] = useState<"approve" | "submit">("approve");
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    if (props.hasApproval) {
+      setView("submit");
```

Comment:
you can do this as initial default value for `useState` rather than a useeffect

Reply:
Done